src/app/state/staking/saga.ts

```diff
@@ -190,6 +190,11 @@ export function* getValidatorDetails({ payload: address }: PayloadAction<string>
 }
 
 export function* fetchAccount({ payload: address }: PayloadAction<string>) {
+  if (!isValidAddress(address)) {
+    yield* put(stakingActions.updateDelegations({ delegations: [], debondingDelegations: [] }))
```

Comment:
Do we need to dispatch this action?

Reply:
We don't strictly need it, but I don't like to idea of keeping data belonging to some previous address.
One could argue that the data won't be displayed anyway, since the account details page is hidden, and if/when a new, correct address will be provided, the new data will be reloaded anyway, and that's all true, but still, there is no need to keep the old, invalid data in memory any longer at this point.

Since the unnecessary line is bothering you, I will get rid of it.